src/openvic-simulation/history/DiplomaticHistory.cpp

```diff
@@ -0,0 +1,311 @@
+#include "DiplomaticHistory.hpp"
+
+#include "openvic-simulation/GameManager.hpp"
+#include "openvic-simulation/dataloader/NodeTools.hpp"
+
+using namespace OpenVic;
+using namespace OpenVic::NodeTools;
+
+WarHistory::added_wargoal_t::added_wargoal_t(
+    Date new_added,
+    Country const* new_actor,
+    Country const* new_receiver,
+    WargoalType const* new_wargoal,
+    std::optional<Country const*> new_third_party,
```

Comment:
Keep in mind that wargoals can also liberate dead countries. Certainly you can make a country object for each tag (or even combine them, such that there is a single object representing both the country and the tag) and instantiate countries for dead tags, but whatever design you end up with, it will also have to accommodate a country changing its tag, and this may cause issues if there is a "dead" country that is already mapped to that tag

Reply:
These structs are used exclusively for history, I believe there will be different structures for general gameplay that will have to eventually take into consideration OOTB mode.

Reply:
Optimally it would be best for us to run through the history data for necessary loading and then delete them asap since they contain a massive amount of wasted memory on things that will never change, pretty much all of the history that isn't being immediately used actually could be cache on a storage drive and we'd only get advantages out of doing so.